desc/basis.py

```diff
@@ -149,16 +149,31 @@ def L(self):
         """int: Maximum radial resolution."""
         return self.__dict__.setdefault("_L", 0)
 
+    @L.setter
+    def L(self, L):
+        assert isinstance(L, int), "Basis Resolution must be an integer!"
```

Comment:
This might be overly restrictive, eg `np.int` wouldn't work here.
A better way might be
```python
assert int(L) == L
self._L = int(L)

Reply:
You may still want to cast it to an int explicitly if it just being a float can cause issues

Ie, ``int(3.0) == 3`` is true